open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/validator/VRPConsentRequestValidator.java

```diff
@@ -0,0 +1,505 @@
+//                            ErrorConstants.INVALID_PARAMETER_AMOUNT,
+//                            ErrorConstants.PATH_MAXIMUM_INDIVIDUAL_AMOUNT);
+//                }
+
+                //validate period alignment
+                if (!ConsentManageUtil.validatePeriodicAlignment(limit)) {
+                    return ConsentManageUtil.getValidationResponse(ErrorConstants.INVALID_REQ_PAYLOAD,
+                            ErrorConstants.INVALID_PERIOD_ALIGNMENT, ErrorConstants.PATH_PERIOD_ALIGNMENT);
+                }
+
+                //validate period type
+                if (!ConsentManageUtil.validatePeriodicType(limit)) {
+                    return ConsentManageUtil.getValidationResponse(ErrorConstants.INVALID_REQ_PAYLOAD,
+                            ErrorConstants.MISSING_PERIOD_TYPE, ErrorConstants.PATH_PERIOD_TYPE);
+                }
+
+                if (StringUtils.isEmpty((String) periodType)) {
+                    return ConsentManageUtil.getValidationResponse(ErrorConstants.INVALID_REQ_PAYLOAD,
+                            ErrorConstants.MISSING_PERIOD_TYPE, ErrorConstants.PATH_PERIOD_TYPE);
+                }
+            }
+            // If all validations pass, set the overall validation status to true
+            validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+        } else {
+            // If periodic limits key is missing, return an error
+            log.error(ErrorConstants.MISSING_PERIOD_LIMITS);
+            return ConsentManageUtil.getValidationResponse(ErrorConstants.INVALID_REQ_PAYLOAD,
+                    ErrorConstants.MISSING_PERIOD_LIMITS,
+                    ErrorConstants.PATH_PERIOD_TYPE);
+        }
+
+        validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+        return validationResponse;
+    }
+
+    public static JSONObject validateParameterDateTime(JSONObject controlParameters) {
+        JSONObject validationResponse = new JSONObject();
+
+        if (controlParameters.containsKey(ConsentExtensionConstants.VALID_TO_DATE_TIME)) {
+
+            // Retrieve the validDateTime from the control parameters
+            Object validateToDateTime = controlParameters.get(ConsentExtensionConstants.VALID_TO_DATE_TIME);
+
+            if (!isValidDateTimeObject(validateToDateTime)) {
+                return ConsentManageUtil.getValidationResponse(ErrorConstants.PAYLOAD_FORMAT_ERROR_VALID_TO_DATE,
+                        ErrorConstants.INVALID_DATE_TIME_FORMAT,
+                        ErrorConstants.PATH_VALID_TO_DATE);
+            }
+
+            // Validate ISO datetime format for validTo
+            if (!isValidISODateTimeFormat((String) validateToDateTime)) {
+                return ConsentManageUtil.getValidationResponse(ErrorConstants.PAYLOAD_FORMAT_ERROR_VALID_TO_DATE,
+                        ErrorConstants.INVALID_DATE_TIME_FORMAT,
+                        ErrorConstants.PATH_VALID_TO_DATE);
+            }
```

Comment:
Why don't we do this validatation inside the isValidDateTimeObject method itself?